### PR TITLE
coalib: Add JSON Schema

### PR DESCRIPTION
**Pull request by nemaniarjun**
Adds JSON Scema to coalib source so
its available to users offline.

Closes https://github.com/coala/coala/issues/2894

### Checklist

- [ ] I have rebased properly. Please see [our tutorial on rebasing](http://coala.io/git#rebasing).
- [ ] I have gone through the [commit guidelines](http://coala.io/commit) and I've followed them.
- [ ] I have followed the [guidelines for the commit shortlog](http://coala.io/commit#shortlog).
- [ ] I have followed the [guidelines for the commit body](http://coala.io/commit#commit-body).
- [ ] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://coala.io/commit#issue-reference).
- [ ] I have [run all the tests](http://api.coala.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### CODE REVIEW

File: coalib/JSONSchema [hunk 1]
```diff
@@ -0,0 +1,119 @@
+
```
Comment (jayvdb) [inline coalib/JSONSchema]: empty line on first line?

File: coalib/JSONSchema [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      },
+      "required": [
+        "default"
+      ]
+    }
+  },
+  "required": [
+    "results"
+  ]
+}
```
Comment (jayvdb) [inline coalib/JSONSchema]: missing EOL at EOF

**Comment by jayvdb** [comment]
The JSON schema file needs an extension, and the JSON bear should be enabled to check this file.

**Comment by nemaniarjun** [comment]
@jayvdb changes made.

File: MANIFEST.in [hunk 3]
```diff
@@ -3,4 +3,5 @@ include LICENSE
 include test-requirements.txt
 include requirements.txt
 include setup.cfg
+include coalib/JSONSchema
```
Comment (jayvdb) [inline MANIFEST.in]: this file doesnt exist

File: setup.py [hunk 4]
```diff
@@ -93,7 +93,7 @@ def run(self):
           packages=find_packages(exclude=['build.*', 'tests', 'tests.*']),
           install_requires=required,
           tests_require=test_required,
-          package_data={'coalib': ['default_coafile', 'VERSION',
+          package_data={'coalib': ['default_coafile', 'VERSION', 'JSONSchema',
```
Comment (jayvdb) [inline setup.py]: this file doesnt exist

Comment (nemaniarjun) [inline setup.py]: ahhh noob mistake :p Will fix

**Comment by nemaniarjun** [comment]
@jayvdb Can we create a different issue for that? Maybe a diff/newcomer

**Comment by sils** [comment]
can we generate the schema? Also can we generate some human readable docs for it ideally even so it appears on api.coala.io?

**Comment by nemaniarjun** [comment]
What do you mean by "generating the schema" ?
Where don't know enough about `coala --json` to write documentation about it.

**Comment by jayvdb** [comment]
The initial basic documentation for a developer feature should be written with the developer feature.
We need to verify that this schema can be used, so an example usage in the docs is the easiest way to do that.

**Comment by nemaniarjun** [comment]
Maybe @sils can help?

**Comment by sils** [comment]
huh there's conflicts

**Comment by nemaniarjun** [comment]
How do we proceed? We should add instructions to the API docs about how to use this.

**Comment by sils** [comment]
I don't want to approve it like this. I want the JSON docs to be generated from the coala output or from the classes or something. This might not be easy (difficulty/medium) but I think this way or generating it once and putting it up in the docs will only lead to everything becoming outdated within short time.

**Pull request closed without merging**